src/platform/Infineon/CYW30739/KeyValueStoreManagerImpl.cpp

```diff
@@ -171,25 +171,27 @@ CHIP_ERROR KeyValueStoreManagerImpl::EraseAll(void)
     return CHIP_NO_ERROR;
 }
 
-KeyValueStoreManagerImpl::KeyStorage::KeyStorage(const char * key, size_t keyLength) : mValueSize(0)
+KeyValueStoreManagerImpl::KeyStorage::KeyStorage(const char * key) : mValueSize(0)
 {
     memset(mKey, 0, sizeof(mKey));
-    memcpy(mKey, key, keyLength);
+
+    if (key != NULL)
+        strncpy(mKey, key, sizeof(mKey));
```

Comment:
This does not guarantee null termination if key `size ==  kKeyLengthMax`. Could this break for you?

Reply:
No, we use `strncmp(mKey, key, sizeof(mKey))` to compare the key.
Also, this change can pass the storage API audit, so I think it works well.